src/wmtk/AttributeCollection.hpp

```diff
@@ -48,6 +48,16 @@ struct AttributeCollection : public AbstractAttributeContainer
         // TODO: in Concurrent, vertex partition id, vertex mutex should be part of attribute
     }
 
+    void shrink_to_fit()
```

Comment:
We'd better not use shrink_to_fit() for attributecollection since it is not threadsafe. Just a suggestion.

Reply:
That function is used in consolidate_mesh. I think it is unavoidable that some functions will not be thread safe. Instead of removing it, I vote for adding documentation and state there that the method is not thread-safe.